core-primitives/test/src/mock/onchain_mock.rs

```diff
@@ -14,20 +14,17 @@ pub struct OnchainMock {
 }
 
 impl OnchainMock {
-	pub fn with_storage_entries<V: Encode>(
-		mut self,
-		entries: Vec<StorageEntryVerified<V>>,
-	) -> Self {
-		for (k, v) in entries.into_iter().map(|e| e.into_tuple()) {
-			self.inner.insert(k, v.map(|v| v.encode()).unwrap());
+	pub fn with_storage_entries<V: Encode>(mut self, entries: Vec<(Vec<u8>, V)>) -> Self {
```

Comment:
Remove the need to expose the otherwise private `StorageEntryVerified` constructor in tests.